Display "Available Funds" in Classroom Students Table

ersync opened the pull request:
### Problem
Teachers/Admins couldn't see how much cash students had available to invest - only portfolio value was shown.

### Changes
* Added "Available Funds" column showing cash balance (teachers/admins only)
* Included `portfolio: :portfolio_transactions` to prevent N+1 queries
* Uses existing `cash_balance` method, no schema changes

### Result
Teachers can now see exactly how much each student has to invest.

### How to test
1. Sign in as teacher/admin and visit `/classrooms/:id`
2. Verify new column shows student cash balances
3. Confirm students can't see this column
4. Check logs for no N+1 queries

Closes #569

davidpgero commented on app/controllers/classrooms_controller.rb:
```diff
@@ -12,7 +12,11 @@ def index
 
   def show
-    @students = @classroom.users.students.includes(:portfolio, :orders)
+    @students = @classroom.users.students.includes(
```
That's good that you're thinking about n+1 queries early! 🏅

davidpgero commented on app/views/classrooms/_classroom_students_table.html.erb:
```diff
@@ -48,6 +51,15 @@
             </td>
+            <% if @can_manage_students %>
+              <td class="px-6 py-4 whitespace-nowrap text-sm text-gray-900 font-medium">
+                <% if student.portfolio %>
```
It looks a little too defensive to me. 🤔 

Since it's a student, we can assume a portfolio already exists. 

https://github.com/rubyforgood/stocks-in-the-future/blob/main/app/models/student.rb#L18-L20

What do you think? We can keep it if it's safer.

davidpgero commented on app/views/classrooms/_classroom_students_table.html.erb:
```diff
@@ -48,6 +51,15 @@
+            <% if @can_manage_students %>
+              <td class="px-6 py-4 whitespace-nowrap text-sm text-gray-900 font-medium">
+                <% if student.portfolio %>
+                  $<%= number_with_precision(student.portfolio.cash_balance, precision: 2) %>
```
I prefer the `number_to_currency`, because it handles the precision, the dollar sign, and it will help us if it needs to work with different currencies later. 

Usage for instance: 

https://github.com/rubyforgood/stocks-in-the-future/blob/main/app/views/stocks/index.html.erb#L16

davidpgero reviewed (commented):
Thank you for working on this one! 

It looks good and is working properly 🚀 

I added a question and a suggestion to this pr.

ersync commented:
> Thank you for working on this one!
> 
> It looks good and is working properly 🚀
> 
> I added a question and a suggestion to this pr.

Thanks for the review! I updated it with `number_to_currency`.

The defensive check was added to match Polly’s same checks in the `Portfolio` and `Portfolio Value` columns. Also since the delegate has `allow_nil`, it seems best to keep them for both consistency and safety.

davidpgero reviewed (approved):
Looks good to me! Thank you!
